fields.go

```diff
@@ -96,3 +144,17 @@ func GetDurationField(r *http.Request, fieldName string) (time.Duration, error)
 	}
 	return d, nil
 }
+
+// Retrieves requested field as an `time.Duration` object, allowSet provides input sanitization. If an error
+// occurs, returns either a `MissingFieldError`, `InvalidFormatError`, or an `UnsafeFieldError`.
+func GetDurationFieldSafe(r *http.Request, fieldName string, allowSet AllowSet) (time.Duration, error) {
```

Comment:
Won't `time.ParseDuration` already filter bad inputs? Same with timestamp and int.